### PR TITLE
fix: Update NumberField to respect readOnly prop.

### PR DESCRIPTION
**Pull request by stephenh**

### CODE REVIEW

File: src/forms/FormStateApp.tsx [hunk 1]
```diff
@@ -130,6 +135,9 @@ export function FormStateApp() {
               />
             </div>
           </header>
+          <div>
+            <Button label="Read Only" onClick={() => setReadOnly(!readOnly)} />
```
Comment (stephenh) [inline src/forms/FormStateApp.tsx:139]: I threw in a button to flip between read only/not read only to see changes...

File: src/forms/StaticField.stories.tsx [hunk 2]
```diff
@@ -15,6 +16,9 @@ export function StaticField() {
         <StaticFieldComponent label="First" value="Bob" />
         <StaticFieldComponent label="First" value="Bob" />
       </FieldGroup>
+      <StaticFieldComponent label="First">
+        <Pills values={["First", "Last"]} />
```
Comment (stephenh) [inline src/forms/StaticField.stories.tsx:20]: @Chieze-Franklin FWIW this PR let's `StaticField` have children that are not "just a string".

File: src/inputs/TextFieldBase.tsx [hunk 3]
```diff
@@ -94,7 +94,16 @@ export function TextFieldBase(props: TextFieldBaseProps) {
           data-readonly="true"
         >
           {multiline
-            ? (inputProps.value as string | undefined)?.split("\n\n").map((p) => <p>{p}</p>)
+            ? (inputProps.value as string | undefined)?.split("\n\n").map((p) => (
+                <p css={Css.my1.$}>
+                  {p.split("\n").map((sentence) => (
```
[the remark this answers is not in the document]
Comment (bdow) [reply src/inputs/TextFieldBase.tsx:99]: Though, now I see we're actually doing a double line break as a paragraph... interesting... I don't know what to call it. 🤷

**Review by bdow — APPROVED** [review]

File: src/inputs/TextFieldBase.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (stephenh) [reply src/inputs/TextFieldBase.tsx:99]: Yeah...agreed it's fuzzy. Going to leave as-is and will fix if something great comes to mind.

**Pull request merged**